src/mender-update/context/context.cpp

```diff
@@ -234,39 +233,52 @@ expected::ExpectedString MenderContext::GetDeviceType() {
 	return expected::ExpectedString(ret);
 }
 
-static error::Error FilterProvides(
+bool CheckClearsMatch(const string &to_match, const string &clears_string) {
+	if (clears_string.empty()) {
+		return to_match.empty();
+	}
+
+	vector<std::string> sub_strings;
+	string escaped;
+	for (const auto chr : clears_string) {
+		if (chr == '*') {
+			sub_strings.push_back(escaped);
+			escaped.clear();
+		} else {
+			escaped.push_back(chr);
+		}
+	}
+	sub_strings.push_back(escaped);
+
+	// Make sure that that front of vector starts at index 0
+	if (sub_strings.front() != "" && to_match.find(sub_strings.front()) != 0) {
+		return false;
+	}
+	// Checks if no trailing wildcard
+	if (sub_strings.back() != ""
+		&& to_match.compare(
+			   to_match.size() - sub_strings.back().size(), to_match.size(), sub_strings.back())
+			   != 0) {
+		return false;
+	}
+
+	for (const auto &str : sub_strings) {
+		if (to_match.find(str) == std::string::npos) {
```

Comment:
I think you need to take into account where the previous sub-match ended, don't you? Otherwise `test_*test_*` would match `test_string`, but it shouldn't.

Reply:
Yes, you're right :)